internal/upstreamldap/upstreamldap.go

```diff
@@ -190,7 +190,12 @@ func (p *Provider) PerformRefresh(ctx context.Context, storedRefreshAttributes p
 	if err != nil {
 		return nil, fmt.Errorf(`error dialing host %q: %w`, p.c.Host, err)
 	}
-	defer conn.Close()
+	defer func(conn Conn) {
```

Comment:
Do you need your func to take an argument here? The local variable `conn` should be captured in the closure of the func, right? I didn't think you need the argument, but `defer` is a subtle beast so I could be remembering wrong.

Reply:
I noticed that too, but I used Goland's `Wrap error handling in a closure` and then just added the `plog` statement assuming that Goland probably had a good default.

Reply:
On second thought, could it be a private helper function so all of these call sites can be one-liners? E.g. `defer closeConn(conn, "refreshing connection")`?

Reply:
@cfryanr done